cli/cmdlineparser.cpp

```diff
@@ -1066,6 +1066,10 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
 #endif
             }
 
+            // Safety certified behavior
+            else if (std::strcmp(argv[i], "--safety") == 0)
```

Comment:
There is no unit test for this option in `TestCmdlineParser`.

There is no documentation either but I do like that because that still allows us to modify this.

I think this should be `--safety-exitcode` (or rather `--critical-exitcode`) instead. That would be more alone options we already have and allows people (rather scripts) to differentiate between the type of the failure.

Reply:
It's not just about exitcode anymore. It's to enable "safety certified behavior". I envision that this option will not be used much except in our testing. Cppcheck Premium users will get a safety configuration in the cppcheck.cfg file.